src/Controller/AppController.php

```diff
@@ -72,8 +73,17 @@ public function initialize() : void
     public function beforeFilter(Event $event) : ?Response
     {
         $tokens = $this->Auth->user('tokens');
-        if ($tokens) {
+        if (!empty($tokens)) {
             $this->apiClient->setupTokens($tokens);
+        } elseif (!in_array($this->request->url, ['login'])) {
+            $route = ['_name' => 'login'];
+            $redirect = Router::reverse($this->request);
```

Comment:
did some tests and redirect url is not working as expected for urls like `/documents/view/123`
But it works with:
```php
            $redirect = $this->request->getUri()->getPath();
            if ($redirect !== $this->request->getAttribute('webroot')) {
```